### PR TITLE
Add API wrappers for existing Clusto types

### PR DESCRIPTION
**Pull request by jasonlai**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -4,7 +4,39 @@ var _ = require('lodash');
...
+module.exports = {
+    CLUSTO_TYPES: CLUSTO_TYPES,
+    CLUSTO_TYPE_PLURALS: CLUSTO_TYPE_PLURALS,
+    ClustoClient: ClustoClient,
+    pluralize: pluralize
+};
+
+function pluralize(word, dictionary) {
+    return _.get(dictionary, word, word + 's');
+}
```
Comment (robskillington) [inline index.js]: Could you move pluralize into a `util.js` to avoid exporting it as the public interface of the module perhaps?

File: index.js [hunk 2]
```diff
@@ -4,7 +4,39 @@ var _ = require('lodash');
...
+var CLUSTO_TYPE_PLURALS = {
+    'networkswitch': 'networkswitches'
+};
+
+module.exports = {
+    CLUSTO_TYPES: CLUSTO_TYPES,
+    CLUSTO_TYPE_PLURALS: CLUSTO_TYPE_PLURALS,
+    ClustoClient: ClustoClient,
+    pluralize: pluralize
+};
```
Comment (robskillington) [inline index.js:36]: To preserve people modifying the constants you could do:

``` js
module.exports = {
    CLUSTO_TYPES: Object.freeze(CLUSTO_TYPES),
    CLUSTO_TYPE_PLURALS: Object.freeze(CLUSTO_TYPE_PLURALS),
    ClustoClient: ClustoClient,
    pluralize: pluralize
};
```

File: index.js [hunk 3]
```diff
@@ -71,3 +103,28 @@ ClustoClient.prototype._tryRequest = function tryRequest(callback, uri, method,
...
+    };
+
+    return this;
+}.bind(ClustoClient);
+
+CLUSTO_TYPES.forEach(function eachClustoType(typeName) {
+    var typeNamePlural = pluralize(typeName, CLUSTO_TYPE_PLURALS);
+
+    ClustoClient.defineClustoType(typeName, typeNamePlural);
+});
```
Comment (robskillington) [inline index.js]: Missing endline here

File: test/index.js [hunk 4]
```diff
@@ -1,9 +1,31 @@
 'use strict';
 
-var ClustoClient = require('../index').ClustoClient;
+var _ = require('lodash');
 var test = require('tape');
 
+var ModNodeClusto = require('../index');
```
Comment (robskillington) [inline test/index.js]: Perhaps just `var clusto =`

File: index.js (hunk 1 above)
Comment (jasonlai) [inline index.js]: Done

File: index.js (hunk 3 above)
Comment (jasonlai) [inline index.js]: Done

File: index.js (hunk 2 above)
Comment (jasonlai) [inline index.js:36]: Legit. Done.

**Comment by robskillington** [comment]
LGTM

**Pull request merged**